### DISCUSSION TITLE
Router: how to know navigation state

### DISCUSSION THREAD

**Discussion by wackazong** [DioxusLabs/dioxus#2368] [Q&A]
I want to add transition functionality to back and forward navigation of the router. Where can I get the info on whether the current route was reached by:

- a Link component
- a back navigation in the browser
- a forward navigation in the browser

Thanks.

**Comment by zoechi** [comment]
https://docs.rs/dioxus-router/latest/dioxus_router/hooks/fn.use_route.html
